xcm/xcm-executor/src/assets.rs

```diff
@@ -216,26 +365,273 @@ impl Assets {
 					// remove the maxmimum possible up to id/amount from self, add the removed onto
 					// result
 					if let Some(entry) = self.non_fungible.take(&(class, instance)) {
-						self.non_fungible.insert(entry);
+						result.non_fungible.insert(entry);
 					}
 				}
-				// TODO: implement partial wildcards.
-				_ => {
-					Default::default()
-				}
-				// MultiAsset::AllFungible
-				// | MultiAsset::AllNonFungible
-				// | MultiAsset::AllAbstractFungible { id }
-				// | MultiAsset::AllAbstractNonFungible { class }
-				// | MultiAsset::AllConcreteFungible { id }
-				// | MultiAsset::AllConcreteNonFungible { class } => (),
 			}
 		}
 		result
 	}
 
+	/// Swaps two mutable Assets, without deinitializing either one.
 	pub fn swapped(&mut self, mut with: Assets) -> Self {
 		swap(&mut *self, &mut with);
```

Comment:
In cases like this, it is not entirely clear (although a seasoned rustecean would have a clue) where this function comes from. In cases like this, people usually import `mem` instead of `mem::swap` and here they'd call `mem::swap(..)`.